Mathlib/Data/Polynomial/Coeff.lean

```diff
@@ -3,10 +3,10 @@ Copyright (c) 2018 Chris Hughes. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Chris Hughes, Johannes Hölzl, Scott Morrison, Jens Wagemaker
 -/
+import Mathlib.Algebra.MonoidAlgebra.Support
+import Mathlib.Algebra.Regular.Pow
```

Comment:
I have the impression this imports quite a lot of stuff. Does it makes to move this result somewhere else?

Reply:
The only new import is `Mathlib.Algebra.MonoidAlgebra.Support`. (The `Regular.Pow` was already in the imports, they were just out of order so it got moved up a couple lines.) `MonoidAlgebra.Support` is only 150 lines, and its only import is `MonoidAlgebra.Basic` which is already being imported through `Polynomial.Basic` anyway. So the new amount of imported material is pretty minimal.

Reply:
Oh, or are you saying that -- unrelated to this PR -- the `Regular` stuff seems to drag in a lot of imports and so maybe should be pulled out somewhere else?

Reply:
I just noticed that `Mathlib.Algebra.Regular.Pow` starts with the sentence "Group powers and other definitions import a lot of the algebra hierarchy", and maybe we don't want `Mathlib.Data.Polynomial.Coeff` import a lot of stuff.

Reply:
I pushed a slight rewriting of the two relevant lemmas. Not really any longer, but it lets it just use Regular.Basic instead of the 'expensive' Regular.Pow